### PR TITLE
B 19845 int

### PR DESCRIPTION
**Pull request by antgmann**
## Summary

This story creates UI elements during the customer's expense reporting process that capture additional sit data, namely the location and storage weight that is included on their receipt. These fields are required when adding a storage expense type on any moving expense. These fields have been added and frontend validation for all fields has been corrected, where there were previously some bugs (read about this more in the testing instructions below). 

> [!TIP]
> This story is downstream of [B-20173](https://github.com/transcom/mymove/pull/12825), and contains it's changes. However, this PR may be easier to review once B-20173 has been merged into integrationTesting.

This story has taken slightly longer to complete due to the issue that B-20173 resolves and the incorrectly working existing frontend validation system, however has still been completed within original estimate time.

## Verification Steps for the Author

These are to be checked by the author.

- [x] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### How to test

1. Access the customer app, create a new user, create and submit a PPM. Details aren't very important here
2. Access the office app, login as a SC, approve the PPM.
3. Access the customer app as your original user, click the add ppm documents button.
4. Continue through the Trip page using dummy data to get to the Expenses page.
5. Once on the expenses page, ensure the UI matches what is outlined in the story's AC when a storage type is selected. It should look something like this: 
image

6. Ensure that the new sit-related fields are required, but only when the type of expense is storage.
7. Submit the expense ticket with all information, and to confirm that it has saved, edit the same expense ticket after to see your data in the input fields.
8. Edit the type of expense ticket to something other than storage, and save it. Ensure it saves properly, and the now-invisible storage fields are not required.
9. Edit the ticket again, and change the type to storage. Note how the storage-related fields are now empty, as the ticket was changed to a type that does not use those fields.

> [!IMPORTANT]  
> Until this ticket, sit-related fields still had frontend validation applying from storage fields even when the type of ticket was not storage, due to a mistaken part of the validation schema. This has been fixed in this ticket, please test it rigorously. 
 

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### CODE REVIEW

**Comment by ajlusk** [comment]
This is related to [some stuff I mentioned in the PR for 20173](https://github.com/transcom/mymove/pull/12825#discussion_r1622410449).

I'm able to submit a value of 0 for the weight stored which is sent a 0 but then converted to, persisted as, and returned as null by backend. I'm also able to submit 0 as an update to existing weight stored value but it doesn't take and retains the original value. This seems less than ideal but not really a significant issue. Per my comment linked above, since the weight stored field is already required by the UI I think adding UI validation requiring the value to be >=1 would prevent these scenarios.

Other than that everything looks and functions correctly.

File: src/components/Customer/PPM/Closeout/ExpenseForm/ExpenseForm.jsx [hunk 1]
```diff
@@ -87,6 +107,42 @@ const ExpenseForm = ({
...
+                          />
+                          <Field
+                            as={Radio}
+                            id="sitLocationDestination"
+                            label="Destination"
+                            name="sitLocation"
+                            value="DESTINATION"
+                            checked={values.sitLocation === 'DESTINATION'}
+                          />
+                          <MaskedTextField
```
Comment (MInthavongsay) [inline src/components/Customer/PPM/Closeout/ExpenseForm/ExpenseForm.jsx:129]: I was  able to enter and save a negative number for the weight. 
Screenshot 2024-06-04 at 10 07 15 AM

Comment (antgmann) [reply src/components/Customer/PPM/Closeout/ExpenseForm/ExpenseForm.jsx:129]: This is strange, signed={false} on 136 should prevent this. I'll look into it more soon.

Comment (antgmann) [reply src/components/Customer/PPM/Closeout/ExpenseForm/ExpenseForm.jsx:129]: Fortunately, thanks to the fix I did for Alex's comment, this is no longer possible to recreate:
image

Comment (ajlusk) [reply src/components/Customer/PPM/Closeout/ExpenseForm/ExpenseForm.jsx:129]: This isn't an issue due to the validation but for a slight better UX, it seems like `min={0}` or `min={1}` would be a better option for disallowing negatives. With `signed={false}` I'm still able to type in a negative sign but not submit it. With the `min` prop I can type in a negative sign at all.

Comment (ajlusk) [reply src/components/Customer/PPM/Closeout/ExpenseForm/ExpenseForm.jsx:129]: `signed={false}` seems to only prevent typing in a negative sign if the `<MaskedTestField>` has a `prefix` prop.

**Review by MInthavongsay — APPROVED** [review]
I can't replicate that negative number anymore after a pull.  Looks good.

**Review by ajlusk — APPROVED** [review]
The issue I mentioned above has been resolved.

**Pull request merged**